File: Data Structures & Algorithms/binary-tree-diameter/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.result = 0
        self.find_diameter(root)


        return self.result
    
    def find_diameter(self, root: TreeNode) -> int:
        if root is None:
            return 0
        
        left = self.find_diameter(root.left)
        right = self.find_diameter(root.right)
        self.result = max(self.result, left + right)

        return max(left, right) + 1

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int GoodNodes(TreeNode root) {
        if (root == null) {
            return 0;
        }

        int result = 0; 
        int max = root.val;       
        CountGoodNodes(root, max, ref result);

        return result;
    }

    public void CountGoodNodes(TreeNode root, int max, ref int result) {
        if (root == null) {
            return;
        }

        if (root.val >= max) {
            result++;
        }

        max = Math.Max(max, root.val);
        CountGoodNodes(root.left, max, ref result);
        CountGoodNodes(root.right, max, ref result);
    }
}

